core/web/webapp.py

```diff
@@ -1,37 +1,94 @@
+)
+api_router.include_router(
+    tasks.router, prefix="/tasks", tags=["tasks"],
+    dependencies=[Depends(auth.get_current_active_user)]
+)
+api_router.include_router(
+    graph.router, prefix="/graph", tags=["graph"],
+    dependencies=[Depends(auth.get_current_active_user)]
+)
+api_router.include_router(
+    templates.router, prefix="/templates", tags=["templates"],
+    dependencies=[Depends(auth.get_current_active_user)]
+)
+api_router.include_router(
+    users.router, prefix="/users", tags=["users"],
+    dependencies=[Depends(auth.get_current_active_user)]
+)
+# Dependencies are set in system endpoints
+api_router.include_router(
+    system.router, prefix="/system", tags=["system"],
 )
-api_router.include_router(entities.router, prefix="/entities", tags=["entities"])
-api_router.include_router(indicators.router, prefix="/indicators", tags=["indicators"])
-api_router.include_router(tag.router, prefix="/tags", tags=["tags"])
-api_router.include_router(tasks.router, prefix="/tasks", tags=["tasks"])
-api_router.include_router(graph.router, prefix="/graph", tags=["graph"])
-api_router.include_router(auth.router, prefix="/auth", tags=["auth"])
-api_router.include_router(templates.router, prefix="/templates", tags=["templates"])
-api_router.include_router(users.router, prefix="/users", tags=["users"])
-api_router.include_router(system.router, prefix="/system", tags=["system"])
 
 app.include_router(api_router, prefix="/api/v2")
+
+async def set_body(request: Request, body: bytes):
+    async def receive() -> Message:
+        return {'type': 'http.request', 'body': body}
+    request._receive = receive
+
+@app.middleware("http")
+async def log_requests(request: Request, call_next):
+    try:
+        req_body = await request.body()
+        await set_body(request, req_body)
+        response = await call_next(request)
+        extra = {
+            "type": "audit.log",
+            "path": request.url.path,
+            "method": request.method,
+            "username": "anonymous",
+            # When behind a proxy, we should start uvicorn with --proxy-headers
+            # and use request.headers.get('x-forwarded-for') instead.
+            "client": request.client.host,
+            "status_code": response.status_code,
+            "content-type": request.headers.get("content-type", ""),
+            "body": b""
+        }
+        if getattr(request.state, 'username', None):
+            extra["username"] = request.state.username
+        if req_body:
+            extra["body"] = req_body
+        if response.status_code == 200:
+            logger.info("Authorized request", extra=extra)
+        elif response.status_code == 401:
+            logger.warning("Unauthorized request", extra=extra)
+        else:
+            logger.error("Bad request", extra=extra)
+        return response
+    except Exception as e:
+        logger.exception("Error while processing request")
+        raise e
```

Comment:
Should we still return the response here to stop logging issues from delivering an otherwise successful response?

Reply:
Fixed in 160031d472d7b0840d0aaca197af13211dc1977d